### PR TITLE
3 switch setup neo4j

### LINKED ISSUE #3: Switch / Setup Neo4j

### PR DESCRIPTION
**Pull request by WhiteDiamondz**

### CODE REVIEW

**Comment by WhiteDiamondz** [comment]
Ah bah le package jwt que j'avais apparemment c'est poubelle xptdrrr

**Comment by WhiteDiamondz** [comment]
That's better

File: routes/users.go [hunk 1]
```diff
@@ -1,51 +1,71 @@
...
-	if insertErr != nil {
-		msg := fmt.Sprintf("user item was not created")
-		c.JSON(http.StatusInternalServerError, gin.H{"error": msg})
-		fmt.Println(insertErr)
-		return
-	}
-
-	c.JSON(http.StatusOK, result)
-}
+// var validate = validator.New()
```
Comment (Brawdunoir) [inline routes/users.go]: Hiiii, ça veut push du code commenté aled.

File: pkg/users/register.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
+package users
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+)
+
+//func(writer http.ResponseWriter, request *http.Request) {
```
Comment (Brawdunoir) [inline pkg/users/register.go]: Code commenté

**Review by Brawdunoir — CHANGES_REQUESTED** [review]
Hey, would be awesome to make this MR as clean and as little as possible, so I suggest :
- Remove commented code
- Re-add gin (without gin, it is not an API anymore ! ;-) ) and the ping/pong GET request
- Remove example (can be stashed on local to serve its purpose for another issue —like CRUD on user)

File: pkg/users/login.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+func CreateToken(user *User) (string, error) {
+	claims := jwt.MapClaims{}
+	claims["authorized"] = true
+	claims["user_id"] = user.Username
+	claims["exp"] = time.Now().Add(time.Minute * 15).Unix()
+	token := jwt.NewWithClaims(jwt.SigningMethodHS512, claims)
+	return token.SignedString([]byte(os.Getenv("SECRET_ACCESS")))
+}
```
Comment (Brawdunoir) [inline pkg/users/login.go]: This example is out of scope of this issue. Add link to the online example in comments on other linked issues ;)

File: pkg/users/register.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+	writer.WriteHeader(201)
+	writer.Header().Add("Content-Type", "application/json")
+	userRegistrationResponse := UserRegistration{
+		User: User{
+			Username: requestUser.Username,
+			Email:    requestUser.Email,
+		}}
+	bytes, _ := json.Marshal(&userRegistrationResponse)
+	_, _ = writer.Write(bytes)
+}
```
Comment (Brawdunoir) [inline pkg/users/register.go]: This example is out of scope of this issue. Add link to the online example in comments on other linked issues ;)

File: pkg/users/repository.go [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+	return err == nil
+}
+
+func hash(password string) (string, error) {
+	hash, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
+	if err != nil {
+		return "", err
+	}
+	return string(hash), nil
+}
```
Comment (Brawdunoir) [inline pkg/users/repository.go]: Same

**Comment by WhiteDiamondz** [comment]
As you probably know gin is not required for the code to be an API !
Without it it can still be one ;)
However, by all means we can use this solution, I'll put it and keep the examples for the corresponding issues

**Review by Brawdunoir — APPROVED** [review]

**Pull request merged**